x-pack/plugin/monitoring/src/test/java/org/elasticsearch/xpack/monitoring/LocalStateMonitoring.java

```diff
@@ -50,6 +54,13 @@ protected SSLService getSslService() {
             protected XPackLicenseState getLicenseState() {
                 return thisVar.getLicenseState();
             }
+
+            @Override
+            public Collection<Module> createGuiceModules() {
+                return XPackPlugin.transportClientMode(settings) ?
+                    Collections.emptyList() :
+                    super.createGuiceModules();
+            }
```

Comment:
Added in order to ignore watcher's feature set object which expects a client to be injected. When creating a transport client, these feature set objects are created at some point and there are no clients available to the injector to give their constructors. This results in a Guice exception.